doc/uml/trait_hierarchy.plantuml

```diff
@@ -10,6 +10,7 @@ abstract class BaseClient {
     ClientResult<()> auto_reauth()
     ClientResult<()> refresh_token()
     String endpoint_url()
+    String auth_url()
```

Comment:
@ramsayleung, can you generate the diagram again? Do you know if there's a way to do this automatically instead?

Reply:
Yes, I can regenerate this diagram, but generating the diagram manually is a little clumsy, I would like to figure out how to generate this diagram automatically.

Reply:
I think we could use this GitHub Action to generate an image from `plantuml` file automatically: https://github.com/marketplace/actions/generate-plantuml

Reply:
Great! Then you can ignore this comment, and we'll set it up in a separate PR.